netbox/netbox/settings.py

```diff
@@ -77,6 +78,8 @@
 EMAIL = getattr(configuration, 'EMAIL', {})
 ENFORCE_GLOBAL_UNIQUE = getattr(configuration, 'ENFORCE_GLOBAL_UNIQUE', False)
 EXEMPT_VIEW_PERMISSIONS = getattr(configuration, 'EXEMPT_VIEW_PERMISSIONS', [])
+GITHUB_REPOSITORY = getattr(configuration, 'GITHUB_REPOSITORY', 'netbox-community/netbox')
+GITHUB_VERSION_TIMEOUT = getattr(configuration, 'GITHUB_VERSION_TIMEOUT', 8 * 3600)
```

Comment:
Per my comment below, let's do validation on `GITHUB_VERSION_TIMEOUT` here. I want to be especially cautious with this setting because it effects calls to an external system.

```suggestion
GITHUB_VERSION_TIMEOUT = int(getattr(configuration, 'GITHUB_VERSION_TIMEOUT', DEFAULT_GITHUB_VERSION_TIMEOUT))

try:
    if GITHUB_VERSION_TIMEOUT and GITHUB_VERSION_TIMEOUT < MINIMUM_GITHUB_VERSION_TIMEOUT:
        raise ImproperlyConfigured("GITHUB_VERSION_TIMEOUT must be at least {} seconds".format(MINIMUM_GITHUB_VERSION_TIMEOUT)
    except ValueError:
        raise ImproperlyConfigured("GITHUB_VERSION_TIMEOUT must be an integer")
```

`DEFAULT_GITHUB_VERSION_TIMEOUT` and `MINIMUM_GITHUB_VERSION_TIMEOUT` should be defined in `extras/constants.py`.

Reply:
IMO `MINIMUM_GITHUB_VERSION_TIMEOUT` should be several (4?) hours. It might make sense to introduce a simple management command to manually force a version check, too. This would be exempt from the minimum timeout.

Reply:
Please also keep an eye on the "rate-limits" of github: https://developer.github.com/v3/#rate-limiting
``For unauthenticated requests, the rate limit allows for up to 60 requests per hour. Unauthenticated requests are associated with the originating IP address, and not the user making requests.``

Reply:
This makes one request every 8 hours, so no problems there